### ISSUE TITLE
Linux snap package O3DE installer: Built projects do not open the Editor and multiple assets fail processing in Asset Processor

### ISSUE THREAD

**Issue by LB-ArturZieba** [o3de/o3de#15856]
**Describe the bug**
If a project is created and built via Project Manager on the snap package O3DE installation, the Editor cannot be launched. Multiple asset processing fails can be seen in the Asset Processor (please refer to the **Screenshots** section to see all of them). Attempting to launch the project's Asset Processor standalone and then launching the Editor does not fix this issue.

It should be noted that already existing projects with processed assets can be launched from the same Project Manager.

**Steps to reproduce**
1. Download O3DE snap package from https://o3debinaries.org/23.05.0/Linux/o3de_23.05.0_amd64.snap.
2. Follow instructions listed in https://github.com/o3de/o3de.org/pull/2385/files for snap package installation.
3. Launch Project Manager as mentioned in the instruction in step 2.
4. Create and build a new project (any template).
5. Open the project.

**Expected behavior**
Editor launches.

**Actual behavior**
Asset processing fail errors are displayed, the Editor does not open, and the Asset Processor is closed.

**Screenshots**
![AP0](https://user-images.githubusercontent.com/86952082/235683588-c3a52252-e553-4ab1-8190-5530693d26a8.png)
![AP1](https://user-images.githubusercontent.com/86952082/235683627-a8666e9a-9915-46a6-8688-b941007d1bf3.png)
![AP2](https://user-images.githubusercontent.com/86952082/235683624-2085a08e-b8de-408f-ba63-b4eaf4040066.png)
![AP3](https://user-images.githubusercontent.com/86952082/235683622-2bc1eb31-f562-4559-8fa1-9c8e0415c487.png)
![AP4](https://user-images.githubusercontent.com/86952082/235683619-f5a9ceb4-5249-4f84-a571-7d75c1457c25.png)
![AP5](https://user-images.githubusercontent.com/86952082/235683616-798274d2-0d41-4209-bd90-0f84644cbc43.png)
![AP6](https://user-images.githubusercontent.com/86952082/235683613-a59ce29b-b2e5-4893-b77f-acaf0e44cb9d.png)
![AP7](https://user-images.githubusercontent.com/86952082/235683608-721d7a0b-d2e3-481d-8e10-e0f42123bd5e.png)
![AP8](https://user-images.githubusercontent.com/86952082/235683606-a1b14486-e931-45ae-bf3f-92c8aa7131d3.png)
![AP9](https://user-images.githubusercontent.com/86952082/235683604-77c56955-4029-4d1e-acfc-ee133c65b703.png)
![AP10](https://user-images.githubusercontent.com/86952082/235683603-8a2d764b-36de-4e1d-ae62-b8b34d91d787.png)
![AP11](https://user-images.githubusercontent.com/86952082/235683600-225b7fc7-39c0-4c80-bb99-22a658709255.png)
![AP12](https://user-images.githubusercontent.com/86952082/235683597-8798dc53-a62e-432b-a822-dc59b87b33c3.png)
![AP13](https://user-images.githubusercontent.com/86952082/235683593-16a0cde3-7838-485d-8525-08027c7641da.png)

**Found in Branch**
https://o3debinaries.org/23.05.0/Linux/o3de_23.05.0_amd64.snap (downloaded 5/2/2023 around 10AM GMT+2)/Stabilization/2305 (215e688)

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
215e688d308ac6ec6b020e5efe95c0aebfce4ced

**Desktop**
 - Device: PC
 - OS: Linux
 - Version: Ubuntu 22.04.1 LTS
 - CPU AMD Ryzen 5 3600
 - GPU NVIDIA GeForce RTX 2060 SUPER
 - Memory 16GB

**Comment by spham-amzn** [comment]
Investigating. So far, was able to catch a stack trace from the editor
```
odule: Attempting to load module:/snap/o3de/x1/23.05.0/bin/Linux/profile/Default/libCrySystem.so
Module: Success!
Process 156704 stopped
* thread #1, name = 'Editor', stop reason = signal SIGINT
    frame #0: 0x00007ffff019aa7c libc.so.6`pthread_kill + 300
libc.so.6`pthread_kill:
->  0x7ffff019aa7c <+300>: movl   %eax, %r13d
    0x7ffff019aa7f <+303>: negl   %r13d
    0x7ffff019aa82 <+306>: cmpl   $0xfffff000, %eax         ; imm = 0xFFFFF000 
    0x7ffff019aa87 <+311>: movl   $0x0, %eax
(lldb) bt
* thread #1, name = 'Editor', stop reason = signal SIGINT
  * frame #0: 0x00007ffff019aa7c libc.so.6`pthread_kill + 300
    frame #1: 0x00007ffff0146476 libc.so.6`raise + 22
    frame #2: 0x00005555560083e8 Editor`___lldb_unnamed_symbol34255 + 24
    frame #3: 0x000055555682c61d Editor`___lldb_unnamed_symbol58458 + 29
    frame #4: 0x000055555682ce20 Editor`___lldb_unnamed_symbol58460 + 1872
    frame #5: 0x00007fff52f5ff67 libAtomFont.so`___lldb_unnamed_symbol60454 + 199
    frame #6: 0x00007fff52f60049 libAtomFont.so`___lldb_unnamed_symbol60455 + 185
    frame #7: 0x00007fff51dda21e libAtomFont.so`___lldb_unnamed_symbol3307 + 46
    frame #8: 0x00007fff51dc85b2 libAtomFont.so`___lldb_unnamed_symbol3118 + 770
    frame #9: 0x00007fff51dcdf84 libAtomFont.so`___lldb_unnamed_symbol3158 + 100
    frame #10: 0x00007fff51dce093 libAtomFont.so`___lldb_unnamed_symbol3159 + 19
    frame #11: 0x00007ffcc6255ed8 libCrySystem.so`___lldb_unnamed_symbol3000 + 456
    frame #12: 0x00007ffcc6241012 libCrySystem.so`___lldb_unnamed_symbol2735 + 2418
    frame #13: 0x00007ffcc62359a5 libCrySystem.so`CreateSystemInterface + 165
    frame #14: 0x00007fffeaa5ddf2 libEditorLib.so`CGameEngine::Init(bool, bool, char const*, IInitializeUIInfo*, void*) + 818
    frame #15: 0x00007fffeac0df7e libEditorLib.so`CCryEditApp::InitGameSystem(void*) + 222
    frame #16: 0x00007fffeac1434e libEditorLib.so`CCryEditApp::InitInstance() + 1262
    frame #17: 0x00007fffeac1a2d7 libEditorLib.so`CryEditMain + 1223
    frame #18: 0x0000555555698530 Editor`___lldb_unnamed_symbol1671 + 224
    frame #19: 0x00007ffff012dd90 libc.so.6`___lldb_unnamed_symbol3139 + 128
    frame #20: 0x00007ffff012de40 libc.so.6`__libc_start_main + 128
    frame #21: 0x0000555555698385 Editor`___lldb_unnamed_symbol1670 + 37
(lldb) exit
```

**Comment by moudgils** [comment]
This feels like something went wrong with AP or AP related builders. Failures where all shaders usually end up being some sort of builders not getting activated.

**Comment by spham-amzn** [comment]
```
~~1683051653955~~1~~0x7f9692ffd640~~AssetBuilder~~ShaderPlatformInterface: Executing '"/snap/o3de/x1/23.05.0/bin/Linux/profile/Default/Builders/DirectXShaderCompiler/bin/dxc" -E MainVS -T vs_6_2 -Zpr -spirv -fvk-use-dx-layout -fvk-invert-y -Fo "/home/ANT.AMAZON.COM/spham/O3DE/Projects/NewProject/user/AssetProcessorTemp/JobTemp-EThvfO/ImagePreview_vulkan.spirv.bin" -Fh "/home/ANT.AMAZON.COM/spham/O3DE/Projects/NewProject/user/AssetProcessorTemp/JobTemp-EThvfO/ImagePreview_vulkan.spirv.txt" "/home/ANT.AMAZON.COM/spham/O3DE/Projects/NewProject/user/AssetProcessorTemp/JobTemp-EThvfO/ImagePreview_vulkan.hlsl.prepend"' ...
~~1683051653955~~1~~0x7f9692ffd640~~AssetBuilder~~C: [Platform API] = vulkan
~~1683051653955~~1~~0x7f9692ffd640~~AssetBuilder~~ProcessWatcher: Child process id 39167 terminated prematurely (exit status 1)
~~1683051653955~~1~~0x7f9692ffd640~~AssetBuilder~~C: [Platform API] = vulkan
~~1683051653956~~1~~0x7f9692ffd640~~AssetBuilder~~C: [Trace] = /data/workspace/o3de/Gems/Atom/RHI/Code/Source/RHI.Edit/Utils.cpp(387): 'bool AZ::RHI::ReportMessages(AZStd::string_view, AZStd::string_view, bool)'
~~1683051653956~~1~~0x7f9692ffd640~~AssetBuilder~~C: [Type] = Trace::Error
```

[ImagePreview.shader-3823292125-5028.log](https://github.com/o3de/o3de/files/11388693/ImagePreview.shader-3823292125-5028.log)

**Comment by spham-amzn** [comment]
It looks like the SNAP package has at least 1 corrupt file: 
```
/snap/o3de/x1/23.05.0/bin/Linux/profile/Default/Builders/DirectXShaderCompiler/bin/dxc
```
This file is not the same size as the one from the 3rd Party package, and attempting to get its version by running
```
./dxc --version
```
Does not report anything

**Comment by AMZN-alexpete** [comment]
> It looks like the SNAP package has at least 1 corrupt file:
> 
> ```
> /snap/o3de/x1/23.05.0/bin/Linux/profile/Default/Builders/DirectXShaderCompiler/bin/dxc
> ```
> 
> This file is not the same size as the one from the 3rd Party package, and attempting to get its version by running
> 
> ```
> ./dxc --version
> ```
> 
> Does not report anything

Good catch, that would definitely break things

**Comment by spham-amzn** [comment]
Update: The dxc file is not corrupt. There is a root issue where all of the 3rd party binaries that are copied has an invalid RPATH value, most likely caused by the PATCHELF process of the snap creation

**Comment by LB-KacperBieniek** [comment]
This issue now also occurs on the latest development snap installer (https://o3debinaries.org/2.0.0/Linux/o3de_2.0.0_amd64.snap) on Ubuntu 22.04. Launching the Editor via Terminal results in the error seen in this issue: https://github.com/o3de/o3de/issues/15837. Here is the full log from launching the Editor with project path provided: [SnapEditorFailsToLaunch.txt](https://github.com/o3de/o3de/files/11429424/SnapEditorFailsToLaunch.txt)

**Comment by LB-ArturZieba** [comment]
Issue still occurs on https://o3debinaries.org/2.0.0/Linux/o3de_2.0.0_amd64.snap installer (downloaded 5/10/2023 around 8:30AM GMT+2), however that could be due to the installer itself not being generated yet. I will reverify this issue again tommorow.

**Comment by LB-ArturZieba** [comment]
Issue still occurs on https://o3debinaries.org/2.0.0/Linux/o3de_2.0.0_amd64.snap installer (downloaded 5/11/2023 around 8:20AM GMT+2). Can you confirm if this is the correct version of the installer for the Development branch?

**Comment by AMZN-alexpete** [comment]
@spham-amzn @LB-ArturZieba I think our nightly installer job has been broken for a while and likely the .snap is not getting updated https://jenkins.build.o3de.org/view/Tools/job/O3DE-development_nightly-installer/  @amzn-changml can you confirm?

**Comment by amzn-changml** [comment]
Last night's snap installer did not complete successfully. I've kicked a new build with no environment variables for version and it produced a proper package. @LB-ArturZieba please try it again. 

There's also a related issue that I've found with Editor opening a new default project in a clean install here: https://github.com/o3de/o3de/issues/15931

**Comment by LB-ArturZieba** [comment]
@amzn-changml  I have checked the https://github.com/o3de/o3de/issues/15931 issue and noticed that the snap installer used in it was 23.05.0 (download: https://o3debinaries.org/23.05.0/Linux/o3de_23.05.0_amd64.snap) as path in one of the lines listed is `/snap/o3de/x1/23.05.0/bin/Linux/profile/Default/libCrySystem.so`, while this issue was reverified on the 2.0.0 (download: https://o3debinaries.org/2.0.0/Linux/o3de_2.0.0_amd64.snap) last two times.

We have verified both of the installers on three machines watching for both this issue and https://github.com/o3de/o3de/issues/15931, and here are our findings:
* On 2.0.0 snap installer this issue has 3/3 reproduction rate still and effectively blocks checking the https://github.com/o3de/o3de/issues/15931 issue.
* On 23.05.0 snap installer, one of the machine reproduced the issue mentioned in the https://github.com/o3de/o3de/issues/15931, while two others encountered https://github.com/o3de/o3de/issues/15939.

Based on the above, am I correct in the assumption that we should compare this issue (and any further snap installer-related ones) against the https://o3debinaries.org/23.05.0/Linux/o3de_23.05.0_amd64.snap installer and not the https://o3debinaries.org/2.0.0/Linux/o3de_2.0.0_amd64.snap as tested twice before?

Thanks!

**Comment by spham-amzn** [comment]
> We noticed that this issue occurs on Development build ([018f149](https://github.com/o3de/o3de/commit/018f14991c4d849a74c8a6cccf1a7cefecb4960a)). We were able to reproduce the issue on Source and Pre-built SDK build. The issue occurs on Ubuntu 20.04. Shaders fail to process in the AssetProcessor if it is ran from the `o3de/build/linux/bin/profile` directory. However, when the AssetProcessor is ran from the project directory the issue occurs only on certain machines. Please refer to the attached DxDiags for more information.
> 
> [DxDiag_project_path_no_fail.txt](https://github.com/o3de/o3de/files/11478556/DxDiag_project_path_no_fail.txt) [DxDiag_project_path_fail.txt](https://github.com/o3de/o3de/files/11478557/DxDiag_project_path_fail.txt)

Why is DxDiag being used to diagnose a Linux issue? The attached logs also indicate that the OS is Windows 11 Pro.

**Comment by LB-MichalRodzos** [comment]
Sorry for the inconvenience. I was unable to gather the hardinfo report due to time constraints. The DxDiag was uploaded to show the hardware the PC (on which the issue was found) is using as the machine has both operating systems. Here are the hardinfo reports with proper details.

[hardinfo_report_project_path_no_fail.txt](https://github.com/o3de/o3de/files/11485778/hardinfo_report_project_path_no_fail.txt)
[hardinfo_report_project_path_fail..txt](https://github.com/o3de/o3de/files/11485780/hardinfo_report_project_path_fail.txt)

**Comment by amzn-changml** [comment]
>Based on the above, am I correct in the assumption that we should compare this issue (and any further snap installer-related ones) against the [o3debinaries.org/23.05.0/Linux/o3de_23.05.0_amd64.snap](https://o3debinaries.org/23.05.0/Linux/o3de_23.05.0_amd64.snap) installer and not the [o3debinaries.org/2.0.0/Linux/o3de_2.0.0_amd64.snap](https://o3debinaries.org/2.0.0/Linux/o3de_2.0.0_amd64.snap) as tested twice before?

These are actually based on two separate branches (2305 = release branch, 2.0.0 = development) so they have to both be tested. The PR based on [#15977](https://github.com/o3de/o3de/pull/15977) is currently going to be integrated into snap packages on both branches, so the result should be in parity for the startup, but not any added features in development

**Comment by LB-ArturZieba** [comment]
Thanks for the response! I will leave this issue closed for now, but will verify again once https://github.com/o3de/o3de/pull/15977 is merged.

**Comment by LB-ArturZieba** [comment]
I have reverified https://o3debinaries.org/2.0.0/Linux/o3de_2.0.0_amd64.snap (downloaded 9:00AM 5/18/2023 GMT+2) and this issue still occurs. https://github.com/o3de/o3de/pull/15977 merge occurred 15 hours ago, so I assume that the installer was not yet updated, I will check this again tomorrow.

**Comment by LB-ArturZieba** [comment]
Issue still occurs on https://o3debinaries.org/2.0.0/Linux/o3de_2.0.0_amd64.snap, tested on two systems at around 9:00AM and 2:00PM GMT+2 5/19/2023. Reopening this issue.

**Comment by spham-amzn** [comment]
This is not a Snap package issue then, but an O3DE issue. I believe this should be a different GHI

**Comment by LB-RobertBaczkowski** [comment]
@spham-amzn An important note: this issue does not occur on the Debian installer, as confirmed today.

**Comment by amzn-changml** [comment]
I've merged in some changes to put the snap in line with the release snap: https://github.com/o3de/o3de/pull/16025. Please try the new snap produced today.

**Comment by LB-ArturZieba** [comment]
We have reverified all .snap installers and the .deb installer today (all downloads were made 5/31/2023 between 8 and 12AM GMT+2 and there were different behaviors noted:

* https://o3debinaries.org/23.05.0/Linux/o3de_23.05.0_amd64.snap - on one PC (Ubuntu 20.04 with NVIDIA GeForce RTX 2060 SUPER/AMD Ryzen 9 3900X configuration running NVIDIA driver metapackage from nvidia-driver-510) Editor was fully functional, while on another https://github.com/o3de/o3de/issues/16047 was encountered.
* https://o3debinaries.org/2.0.0/Linux/o3de_2.0.0_amd64.snap - on both PCs this issue still occurs.
* https://o3debinaries.org/2.1.0/Linux/o3de_2.1.0_amd64.snap - on one PC (Ubuntu 22.04.2 LTS; AMD Ryzen 5 3600X 6-Core Processor/AMD Radeon RX 5700 XT; Memory 16GB) Editor was fully functional, while on another https://github.com/o3de/o3de/issues/15939#issuecomment-1569986172 was encountered.
* https://o3debinaries.org/development/Latest/Linux/O3DE_latest.deb - this was tested on one PC (AMD Ryzen 5 3600X 6-Core Processor/AMD Radeon RX 5700 XT one) and Editor was fully functional.

In short, the issue is still active, but the behavior is inconsistent between different installers and PCs. We will reverify it again the next day on the off-chance that downloaded installers were not updated yet.

**Comment by LB-ArturZieba** [comment]
I have reverified the above today (6/1/2023 downloads around 8-9:00AM GMT+2) on .snap 23.05.0/2.0.0/2.1.0 installers and the behavior described still occurs.

**Comment by amzn-changml** [comment]
I've regenerated the snap for 2.1.0 with the latest code changes in the development branch. @LB-ArturZieba please recheck this one. 

2.0.0 is no longer tracking any branches (it used to be the stabilization branch), so we can exclude this from further testing.

**Comment by LB-ArturZieba** [comment]
I was able to launch the Editor as before on the 2.1.0 .snap installer (downloaded 6/2/2023 8:20AM GMT+2), however https://github.com/o3de/o3de/issues/16047 occurred on it, as described in https://github.com/o3de/o3de/issues/16047#issuecomment-1573244993.

**Comment by spham-amzn** [comment]
Closing issue as fixed, https://github.com/o3de/o3de/issues/16047 is a different issue
